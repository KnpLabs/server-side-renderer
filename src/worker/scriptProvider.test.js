import createScriptProvider from './scriptProvider'
import fs from 'fs'

jest.mock('fs')

beforeEach(() => {
  fs.readFileSync.mockClear()
})

describe('worker :: scriptProvider', () => {
  it(`throws an exception when requesting a non supported script`, () => {
    const scriptProvider = createScriptProvider()

    expect(() => scriptProvider.get('nonSupportedScript')).toThrowErrorMatchingSnapshot()
  })

  it(`throws an exception when the script content is invalid`, () => {
    const scriptProvider = createScriptProvider()

    fs.readFileSync.mockReturnValueOnce('notTheRightVariable = () => {};')

    expect(() => scriptProvider.get('postRender')).toThrowErrorMatchingSnapshot()
  })

  it(`returns the postRender script`, () => {
    const scriptProvider = createScriptProvider()

    fs.readFileSync.mockReturnValueOnce('script = () => {};')

    expect(scriptProvider.get('postRender')).toEqual(expect.any(Function))
  })
})
