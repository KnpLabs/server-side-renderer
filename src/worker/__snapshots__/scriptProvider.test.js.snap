// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`worker :: scriptProvider throws an exception when requesting a non supported script 1`] = `"Non supported script key. \\"nonSupportedScript\\" provided but expected one of postRender."`;

exports[`worker :: scriptProvider throws an exception when the script content is invalid 1`] = `"The script provided doesn't set the script variable as a valid function."`;
